planner/logical_test.go

```diff
@@ -125,3 +129,196 @@ func TestFluxSpecToLogicalPlan(t *testing.T) {
 		})
 	}
 }
+
+type MergeFiltersRule struct {
+}
+
+func (MergeFiltersRule) Name() string {
+	return "mergeFilters"
+}
+
+func (MergeFiltersRule) Pattern() planner.Pattern {
+	return planner.Pat(transformations.FilterKind,
+		planner.Pat(transformations.FilterKind,
+			planner.Any()))
+}
+
+func (MergeFiltersRule) Rewrite(pn planner.PlanNode) (planner.PlanNode, bool) {
+	specTop := pn.ProcedureSpec()
+
+	filterSpecTop := specTop.(*transformations.FilterProcedureSpec)
+	filterSpecBottom := pn.Predecessors()[0].ProcedureSpec().(*transformations.FilterProcedureSpec)
+	mergedFilterSpec := mergeFilterSpecs(filterSpecTop, filterSpecBottom)
+
+	return planner.MergeLogicalPlanNodes(pn, pn.Predecessors()[0], mergedFilterSpec), true
+}
+
+func mergeFilterSpecs(a, b *transformations.FilterProcedureSpec) planner.ProcedureSpec {
+	fn := a.Fn.Copy().(*semantic.FunctionExpression)
+
+	aExp, aOK := a.Fn.Body.(semantic.Expression)
+	bExp, bOK := b.Fn.Body.(semantic.Expression)
+
+	if !aOK || !bOK {
+		panic("function body not expression")
```

Comment:
So does this mean return statements are not allowed?

Reply:
Yes.  I was not thorough since this is just a unit test.  I'll add a comment to that effect.